src/java/org/apache/ivy/plugins/report/XmlReportParser.java

```diff
@@ -71,132 +71,139 @@
 
```

Comment:
This file has a large change which appears to be reordering/restructuring the code logic. Given the nature of this diff, it's difficult to review it and see if this change introduces any changes in the logic or introduces any regressions. Plus this class is one of the very core ones. Also, I'm not too sure if the existing test cases cover for this level of change.

Given all this, I'm not confident that these changes in this specific class are worth it, at least at the moment. Unless of course, some one else can review this specific class and approve it, would you be willing to undo the changes to this one class in this PR?